File: src/main/kotlin/network/bobnet/cms/model/content/Page.kt
package network.bobnet.cms.model.content

import network.bobnet.cms.model.data.RenderedPage
import network.bobnet.cms.model.user.User
import java.time.LocalDateTime
import javax.persistence.*

@Entity
data class Page(var title: String = "",
                var headline: String = "",
                @Column(columnDefinition = "LONGTEXT")
                var content: String = "",
                var featuredImage: String = "",
                @ManyToOne(cascade = [CascadeType.ALL]) var author: User = User(),
                var slug: String = "",
                var addedAt: LocalDateTime = LocalDateTime.now(),
                @Id @GeneratedValue var id: Long? = null,){

    fun render() = id?.let {
        RenderedPage(
                slug,
                title,
                headline,
                content,
                featuredImage,
                author,
                addedAt,
                it
        )
    }
}

File: backend/src/entity/content/page.ts
// @ts-ignore
import { Schema, Model } from "mongoose";
import {Long} from "long";

module.exports = Model('Page', new Schema({
    title: String,
    headline: String,
    content: String,
    featuredImage: String,
    author: Object,
    slug: String,
    addedAt: Date,
    id: Long
}));